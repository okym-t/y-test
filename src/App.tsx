import { css } from '@emotion/react'
import { useEffect, useState } from 'react'

const styles = {
  foo: css({
    width: '100%',
    padding: '20px',
    backgroundColor: 'red',
  }),
  bar: css({
    color: 'blue',
    textAlign: 'center',
  }),
}

const URL = import.meta.env.VITE_API_URL ?? ''

function App() {
  const [prefectures, setPrefectures] = useState<any | undefined>()
  useEffect(() => {
    fetch(`${URL}/api/v1/prefectures`)
      .then((response) => response.json())
      .then((json) => setPrefectures(json))
  }, [])

  if (!prefectures) return <p>loading...</p>
  return (
    <div css={styles.foo}>
      <p css={styles.bar}>{JSON.stringify(prefectures)}</p>
    </div>
  )
}

export default App
