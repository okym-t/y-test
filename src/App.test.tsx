import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import App from './App'

test.skip('should render correctly', () => {
  const { container } = render(<App />)
  expect(container).toMatchSnapshot()
})
