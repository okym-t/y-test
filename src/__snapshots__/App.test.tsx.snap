// Vitest Snapshot v1

exports[`should render correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="/src/logo.svg"
      />
      <p>
        Hello Vite + React!
      </p>
      <p>
        <button
          type="button"
        >
          count is: 
          0
        </button>
      </p>
      <p>
        Edit 
        <code>
          App.tsx
        </code>
         and save to test HMR updates.
      </p>
      <p>
        <a
          class="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
         | 
        <a
          class="App-link"
          href="https://vitejs.dev/guide/features.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vite Docs
        </a>
      </p>
    </header>
  </div>
</div>
`;
